### ISSUE TITLE
Gracefully handle missing elements

### ISSUE THREAD

**Issue by jwestbrook** [sstephenson/prototype#296]
Scenario

```javascript
document.observe('dom:load',function(){

   $('div1').update('my div')

});
```
If the element `div1` does not exist in the DOM then the js errors and nothing else past that line is ran. The error ends up being `cannot read property of null` which is not descriptive. (could be different in different browsers but still not the actual problem)

so question is
1. This is expected behavior, devs should wrap the statements in if blocks
2. We should throw a better error, "Element with id(s) `div1` does not exist"
3. Gracefully fail, or return an empty element to prevent the JS error and throw a warning

This applies to `Element#down()` as well

**Comment by jwestbrook** [comment]
I agree that I want to know that an element is missing, sometimes tracking it in minified or uglified code is difficult though and the custom throw would be useful to narrow it down what `$()` is barfing.

Currently its a cryptic error anyway so throwing a specific error would be an improvement, with minimal effort, and match expectations - ie if I give `$()` a bad id it will tell me, instead of surprise I get an error about not being able to read a property.

I know you are close to getting the next bugfix out - just wanted to see if a concern one of the devs I know could be addressed.

**Comment by savetheclocktower** [comment]
This might be worth doing when a single string is passed into `$` just to make debugging a bit easier. I'll put it in the possible enhancements list.
